DataDog/datadog-operator
# [profiles] Add event filter for watching dap

### What does this PR do?

Add an event filter to send DDA reconcile requests only when a DAP generation changes.

### Motivation

Increased reconciles after adding DAP status since every DAP change was causing a DDA reconcile request.
https://datadoghq.atlassian.net/browse/CECO-1219

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: n/a
* Cluster Agent: n/a

### Describe your test plan

Staging:
* Deploy the operator and check that the number of reconciles decreases

Other:
* Deploy an operator and spin up a DDA and profile
* Look at the number of reconciles. You can use the timestamp and `Reconciling DatadogAgent` log in the operator logs or the metric `datadog.operator.controller_runtime_reconcile_total` with `controller:datadogagent`. With default settings, it should be around 1 reconcile per 15 sec
* Edit the profile so that it won't change the generation (edit the status or add a label)
* Check that the profile generation remains the same (`kubectl describe dap <profile>`) and that the datadogagent reconcile count doesn't change per time period since the generation hasn't changed
* Edit the profile to change its generation (edit the profile spec)
* Check that the profile generation increases and that there's an extra datadogagent reconcile

### Checklist
